addons/point_of_sale/models/pos_order.py

```diff
@@ -600,61 +600,62 @@ def _create_invoice(self, move_vals):
+                                'debit': rounding_applied < 0.0 and -rounding_applied or 0.0,
+                                'credit': rounding_applied > 0.0 and rounding_applied or 0.0,
+                                'account_id': account_id,
+                                'price_unit': rounding_applied,
+                            })
+
+                    else:
+                        self.env['account.move.line'].with_context(skip_invoice_sync=True).create({
+                            'balance': -rounding_applied,
+                            'quantity': 1.0,
+                            'partner_id': new_move.partner_id.id,
+                            'move_id': new_move.id,
+                            'currency_id': new_move.currency_id.id,
+                            'company_id': new_move.company_id.id,
+                            'company_currency_id': new_move.company_id.currency_id.id,
+                            'display_type': 'rounding',
+                            'sequence': 9999,
+                            'name': self.config_id.rounding_method.name,
                             'account_id': account_id,
-                            'price_unit': rounding_applied,
                         })
-
                 else:
-                    self.env['account.move.line'].with_context(skip_invoice_sync=True, check_move_validity=False).create({
-                        'balance': -rounding_applied,
-                        'quantity': 1.0,
-                        'partner_id': new_move.partner_id.id,
-                        'move_id': new_move.id,
-                        'currency_id': new_move.currency_id.id,
-                        'company_id': new_move.company_id.id,
-                        'company_currency_id': new_move.company_id.currency_id.id,
-                        'display_type': 'rounding',
-                        'sequence': 9999,
-                        'name': self.config_id.rounding_method.name,
-                        'account_id': account_id,
+                    if rounding_line:
+                        rounding_line.with_context(skip_invoice_sync=True).unlink()
+                if rounding_line_difference:
+                    existing_terms_line = new_move.line_ids.filtered(
+                        lambda line: line.account_id.account_type in ('asset_receivable', 'liability_payable'))
+                    if existing_terms_line.debit > 0:
+                        existing_terms_line_new_val = float_round(
+                            existing_terms_line.debit + rounding_line_difference,
+                            precision_rounding=new_move.currency_id.rounding)
+                    else:
+                        existing_terms_line_new_val = float_round(
+                            -existing_terms_line.credit + rounding_line_difference,
+                            precision_rounding=new_move.currency_id.rounding)
```

Comment:
unrelated to the pr itself but it triggers me :laughing: 
```suggestion
                    balance = existing_terms_line.debit or -existing_terms_line.credit
                    existing_terms_line_new_val = float_round(
                        balance + rounding_line_difference,
                        precision_rounding=new_move.currency_id.rounding)
```